paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/GooglePayButton.kt

```diff
@@ -46,23 +69,99 @@ internal class GooglePayButton @JvmOverloads constructor(
         // Call super so we don't inadvertently effect the primary button as well.
         super.setClickable(true)
         super.setEnabled(true)
-        viewBinding.googlePayPrimaryButton.backgroundTintList = null
-        viewBinding.googlePayPrimaryButton.finishedBackgroundColor = Color.TRANSPARENT
+    }
+
+    fun initialize(
+        cornerRadius: Int,
+        allowCreditCards: Boolean,
+        billingAddressParameters: GooglePayJsonFactory.BillingAddressParameters?
+    ) {
+        initializePrimaryButton()
+        val isDark = context.isSystemDarkTheme()
+        val allowedPaymentMethods = GooglePayJsonFactory(context)
+            .createIsReadyToPayRequest(
+                billingAddressParameters = billingAddressParameters,
+                allowCreditCards = allowCreditCards
+            ).getJSONArray(ALLOWED_PAYMENT_METHODS).toString()
```

Comment:
Can we make this a method in `GooglePayJsonFactory` instead? We already have `createCardPaymentMethod` which could possibly be made library-internal.